src/bond.py

```diff
@@ -63,3 +63,100 @@ def __init__(self):
             ),
         )
         self._add_typeparam(params)
+
+
+class Quartic(bond.Bond):
+    r"""Quartic bond potential.
+
+    `Quartic` specifies a quartic potential between the two particles in
+    the simulation state with:
+
+    .. math::
+        \begin{eqnarray*}
+        U(r) = & k [(r - r_{0}) - b_{1}][(r - r_{0}) - b_{2}](r - r_{0})^2
+                  + U_{0} + U_{\rm WCA}(r)      & \text{when } r < r_0\\
+             = & U_0                            & \text{when } r\ge r_0
+        \end{eqnarray*}
+
+    where :math:`\vec{r}` is the vector pointing from one particle to the other in the
+    bond.
+    The potential :math:`U_{\rm WCA}(r)` is given by:
+
+    .. math::
+        :nowrap:
+
+        \begin{eqnarray*}
+        U_{\mathrm{WCA}}(r)= & 4 \varepsilon \left[ \left( \frac{\sigma}{r} \right)^{12}
+                             - \left( \frac{\sigma}{r} \right)^{6} \right] + \varepsilon
+                                          & \text{when } r < 2^{\frac{1}{6}}\sigma\\
+                           = & 0          & \text{when } r \ge 2^{\frac{1}{6}}\sigma
+        \end{eqnarray*}
+
+
+    Attributes:
+        params (TypeParameter[``bond type``, dict]):
+            The parameters of the quartic bonds for each particle type.
+            The dictionary has the following keys:
+
+            * ``k`` (`float`, **required**) - quartic attractive force strength
+              :math:`[\mathrm{energy/length^4}]`
+
+            * ``r_0`` (`float`, **required**) - Location of the quartic potential
+              cutoff. Intended to be larger than the WCA cutoff,
+              :math:`2^{\frac{1}{6}}\sigma`. When true,
+              :math:`U(r_0) = U_{0} + U_{\rm WCA}(r_0)`
+              :math:`[\mathrm{length}]`
+
+            * ``b_1`` (`float`, **required**) - First quartic potential fitting
+              parameter
+              :math:`[\mathrm{length}]`
+
+            * ``b_2`` (`float`, **required**) - Second quartic potential fitting
+              parameter
+              :math:`[\mathrm{length}]`
+
+            * ``U_0`` (`float`, **required**) - Quartic potential energy barrier height
+              at :math:`r_0` when :math:`r_0 > 2^{\frac{1}{6}}\sigma`
+              :math:`[\mathrm{energy}]`
+
+            * ``epsilon`` (`float`, **required**) - Repulsive WCA force strength
+              parameter
+              :math:`[\mathrm{energy}]`
+
+            * ``sigma`` (`float`, **required**) - Repulsive WCA interaction distance
+              parameter
+              :math:`[\mathrm{length}]`
+
+            * ``Delta`` (`float`, **optional**) - bond length shift ``Delta``.
+              Effectively replaces all instances of :math:`r \text{ with } r - \Delta`.
+              The default value is zero.
+              :math:`[\mathrm{length}]`
+
+    Examples::
+
+        quartic = azplugins.bond.Quartic()
+        quartic.params['A-A'] = dict(k=1434.3, r_0=1.5, b_1=-0.7589, b_2=0.0,
```

Comment:
These are a rather random set of parameters. Do they come from somewhere that should be referenced / explained, or can they be simplified? Also, I know Antonia mentioned having only one example, but I would not use `delta=0` in the example if is the default. You can omit it, make this example use a nonzero delta, or make two examples (one with default delta, one without).

Reply:
These parameters come from a 2004 paper by Tsige and Stevens, who arrived at these parameters by fitting a FENE bond potential to this quartic bond potential form [here](https://pubs.acs.org/doi/10.1021/ma034970t) I chose to use them here since this paper was one of the first to use this potential in the context of simulating breakable bonds with MD. They can, however, just as easily be replaced by "nicer" quantities. Which would you prefer?

Thank you for the note on the examples! I'll make two examples then.

Reply:
OK! Please add some text and the DOI link to that paper to briefly explain this. Context is helpful for new folks reading documentation.